python-scripts/xbox/interface/if_xbdm.py

```diff
@@ -77,21 +81,32 @@ def xbdm_command(cmd, length=None):
   return lines
 
 def GetModules():
-  return [] # FIXME: Make this work..
+  modulesList = []
   lines = xbdm_command("modules")
-  for line in lines:
-    print(line)
-    chunks = line.split()
-    for chunk in chunks:
-      if (chunk[0:6] == b'name=\"'):
-        print("  name = '" + str(chunk[6:-1]) + "'")
-      elif (chunk[0:5] == b'base='):
-        print("  base = '" + str(chunk[5]) + "'")
-      else:
-        print('  unparsed: ' + str(chunk))
-        pass
-        #assert(False)
-  return []
+
+  linesList = lines
+
+  for line in linesList: 
+
+    nameAddrDict ={}
+    line = line.decode("utf-8") 
+
+    nameAddrChunks = line.split()
+
+    for nameAddrChunk in nameAddrChunks:
+       singleChunk = nameAddrChunk.split("=")    #name="xbdm.dll" ->  ['name', '"xbdm.dll"']
+
+       if(singleChunk[0] =="name"):  
```

Comment:
Please check for this by testing if the first symbol is `singleChunk[1][0] == '"'` (and possibly asserting that the last element `singleChunk[1][-1]` is also the matching quote symbol).

Other multi line responses use a similar format and keys which are not "name" will also contain strings